### DISCUSSION TITLE
How to automatically translate?

### DISCUSSION THREAD

**Discussion by aoeng** [Laravel-Lang/common#162] [Q&A] [answered]
eg: add `message.php` to `zh_CN` 

```
return [
    'error' => '系统繁忙'
];
```

How to automatically make file `message.php` to `en` and other ? And translate `error`?

**Comment by andrey-helldar** [comment]
If this line is in your package, which you created using our [template](https://github.com/Laravel-Lang/translations-template) or something similar, then you can perform the translation by calling the console command:

```bash
# will run translation only for zh_CN locale
vendor/bin/lang translate zh_CN

# will run tranlations for all installed locales into your package
vendor/bin/lang translate
```

> Note
>
> Under the hood, Google Translate is used, which does not translate into all languages, so it is not recommended to run the command without an argument.
> 
> The list of available localizations can be found [here](https://github.com/Laravel-Lang/.github/blob/main/.github/workflows/translate.yml#L13).

And if this file is located in your Laravel project, then our current software solutions cannot help you with the translation. In this case, we recommend using online translators, for example, Google Translate.

Or to translate your phrases, you can use third-party packages, for example, [stichoza/google-translate-php](https://github.com/Stichoza/google-translate-php).

**Reply by aoeng** [reply]
In laravel project,Seems to have no mature solution?
eg :
```
php artisan lang:translate zh_CN
```
I think this is a common problem during development

**Reply by andrey-helldar** [reply, marked as answer]
Laravel Lang projects do not have this, but you can use another package from a third-party developer: [tanmuhittin/laravel-google-translate](https://github.com/tanmuhittin/laravel-google-translate)

```bash
composer require tanmuhittin/laravel-google-translate

php artisan vendor:publish --provider="Tanmuhittin\LaravelGoogleTranslate\LaravelGoogleTranslateServiceProvider"
```

Next, [configure](https://github.com/tanmuhittin/laravel-google-translate/blob/master/src/config/laravel_google_translate.php) it, after which you can translate files:

```bash
php artisan translate:files
```

**Reply by vcardillo** [reply]
Thanks for sharing this. It's a great tool.

**Reply by andrey-helldar** [reply]
You're welcome
